test/e2e/broker_event_transformation_test.go

```diff
@@ -0,0 +1,157 @@
+             1     v	 2     | v        3           |
+EventSource ---> Broker ---> Trigger1 -------> Service(Transformation)
+                   |
+                   | 6                   7
+                   |-------> Trigger2 -------> Service(Logger)
+
+Note: the number denotes the sequence of the event that flows in this test case.
+*/
+
+func TestEventTransformationForTrigger(t *testing.T) {
+	const (
+		brokerName  = "e2e-eventtransformation-broker"
+		rbacObjName = "eventing-broker-filter"
+
+		any          = v1alpha1.TriggerAnyFilter
+		eventType1   = "type1"
+		eventType2   = "type2"
+		eventSource1 = "source1"
+		eventSource2 = "source2"
+		eventBody    = "e2e-eventtransformation-body"
+
+		triggerName1 = "trigger1"
+		triggerName2 = "trigger2"
+
+		transformationPodName = "trans-pod"
+		loggerPodName         = "logger-pod"
+	)
+
+	clients, ns, provisioner, cleaner := Setup(t, true, t.Logf)
+	defer TearDown(clients, ns, cleaner, t.Logf)
+
+	// creates ServiceAccount and ClusterRoleBinding with default cluster-admin role
+	err := CreateServiceAccountAndBinding(clients, rbacObjName, ns, t.Logf, cleaner)
+	if err != nil {
+		t.Fatalf("Failed to create the ServiceAccount and ServiceAccountRoleBinding: %v", err)
+	}
+
+	// create a new broker
+	broker := test.Broker(brokerName, ns, test.ClusterChannelProvisioner(provisioner))
+	t.Logf("provisioner name is: %s", broker.Spec.ChannelTemplate.Provisioner.Name)
+	err = WithBrokerReady(clients, broker, t.Logf, cleaner)
+	if err != nil {
+		t.Logf("Broker is: %v", broker)
+		t.Fatalf("Error waiting for the broker to become ready: %v", err)
+	}
+	brokerUrl := fmt.Sprintf("http://%s", broker.Status.Address.Hostname)
+	t.Logf("The broker is ready with url: %q", brokerUrl)
+
+	// create an event we want to send
+	eventToSend := &test.CloudEvent{
+		Source:   eventSource1,
+		Type:     eventType1,
+		Data:     fmt.Sprintf(`{"msg":%q}`, eventBody),
+		Encoding: test.CloudEventDefaultEncoding,
+	}
+
+	// create the event we want to tranform to
+	transformedEventBody := fmt.Sprintf("%s %s", eventBody, string(uuid.NewUUID()))
+	eventAfterTransformation := &test.CloudEvent{
+		Source:   eventSource2,
+		Type:     eventType2,
+		Data:     fmt.Sprintf(`{"msg":%q}`, transformedEventBody),
+		Encoding: test.CloudEventDefaultEncoding,
+	}
+
+	// create the transformation pod and service, and get them ready
+	transformationPodSelector := map[string]string{"e2etest": string(uuid.NewUUID())}
+	transformationPod := test.EventTransformationPod(transformationPodName, ns, transformationPodSelector, eventAfterTransformation)
+	transformationSvc := test.Service(transformationPodName, ns, transformationPodSelector)
+	transformationPod, err = CreatePodAndServiceReady(clients, transformationPod, transformationSvc, t.Logf, cleaner)
+	if err != nil {
+		t.Fatalf("Failed to create transformation pod and service, and get them ready: %v", err)
+	}
+
+	trigger1 := test.NewTriggerBuilder(triggerName1, ns).
```

Comment:
could htis be done in a range? instead of trigger1, 2 etc ?

Reply:
Ditto.